### PR TITLE
refactor: improve internal functions of `ERC725XCore`

### PR DESCRIPTION
**Pull request by CJ42**
# What does this PR introduce?

- Improve the internal functions of `ERC725XCore.sol`, so to allow better and easier overriding of the `execute(...)` function.
- mark the `internal` functions as `virtual`, so they can be overridden individually.

> made in peer programming with @YamenMerhi 🤝 using [VS Code Live Share](https://marketplace.visualstudio.com/items?itemName=MS-vsliveshare.vsliveshare-pack) extension.

### CODE REVIEW

**Review by b00ste — APPROVED** [review]

File: implementations/contracts/ERC725XCore.sol [hunk 1]
```diff
@@ -33,184 +33,180 @@ import {
...
         result = Address.verifyCallResult(success, returnData, "ERC725X: Unknown Error");
+
+        emit Executed(OPERATION_DELEGATECALL, to, value, bytes4(data));
     }
 
     /**
      * @dev perform contract creation using operation 1
-     * Taken from GnosisSafe: https://github.com/gnosis/safe-contracts/blob/main/contracts/libraries/CreateCall.sol
-     *
+     * @param to The recipient address passed to execute(...) (MUST be address(0) for CREATE2)
```
Comment (YamenMerhi) [inline implementations/contracts/ERC725XCore.sol:157]: MUST be address(0) for CREATE

File: implementations/contracts/ERC725XCore.sol [hunk 2]
```diff
@@ -33,184 +33,180 @@ import {
...
      * @param value The value to be sent to the contract created
-     * @param deploymentData The contract bytecode to deploy
+     * @param data The contract bytecode to deploy
      * @return newContract The address of the contract created
```
Comment (YamenMerhi) [inline implementations/contracts/ERC725XCore.sol:160]: The address of the contract created as bytes

File: implementations/contracts/ERC725XCore.sol [hunk 3]
```diff
@@ -33,184 +33,180 @@ import {
...
+        emit ContractCreated(OPERATION_CREATE, contractAddress, value);
+    }
 
     /**
-     * @inheritdoc ERC165
+     * @dev perform contract creation using operation 2
+     * @param to The recipient address passed to execute(...) (MUST be address(0) for CREATE2)
+     * @param value The value to be sent to the contract created
+     * @param data The contract bytecode to deploy
+     * @return newContract The address of the contract created
```
Comment (YamenMerhi) [inline implementations/contracts/ERC725XCore.sol:191]: The address of the contract created as bytes

File: implementations/contracts/ERC725XCore.sol [hunk 4]
```diff
@@ -33,184 +33,180 @@ import {
...
+        newContract = abi.encodePacked(contractAddress);
+        emit ContractCreated(OPERATION_CREATE, contractAddress, value);
+    }
 
     /**
-     * @inheritdoc ERC165
+     * @dev perform contract creation using operation 2
+     * @param to The recipient address passed to execute(...) (MUST be address(0) for CREATE2)
+     * @param value The value to be sent to the contract created
+     * @param data The contract bytecode to deploy
```
Comment (YamenMerhi) [inline implementations/contracts/ERC725XCore.sol:190]: The contract bytecode to deploy combined with the salt

File: implementations/contracts/ERC725XCore.sol [hunk 5]
```diff
@@ -33,184 +33,180 @@ import {
...
+    {
+        return interfaceId == _INTERFACEID_ERC725X || super.supportsInterface(interfaceId);
     }
 
     /* Internal functions */
 
     /**
      * @dev perform call using operation 0
-     * Taken from GnosisSafe: https://github.com/gnosis/safe-contracts/blob/main/contracts/base/Executor.sol
      *
```
Comment (YamenMerhi) [inline implementations/contracts/ERC725XCore.sol:93]: Let's remove this empty line (consistency)

**Review by YamenMerhi — APPROVED** [review]

**Pull request merged**